ocamltest/ocaml_actions.ml

```diff
@@ -450,11 +459,51 @@ let ocamlopt_opt = Actions.make
   "ocamlopt.opt"
   (compile_test_program Builtin_variables.program2 ocamlopt_opt_compiler)
 
+let run_expect_once ocamlsrcdir input_file principal log env =
+  let expect_flags = try Sys.getenv "EXPECT_FLAGS" with Not_found -> "" in
+  let repo_root = "-repo-root " ^ ocamlsrcdir in
+  let principal_flag = if principal then "-principal" else "" in
+  let commandline =
+  [
+    expect_command ocamlsrcdir;
+    expect_flags;
+    flags env;
+    repo_root;
+    principal_flag;
+    input_file
+  ] in
+  let exit_status = Actions_helpers.run_cmd log env commandline in
+  if exit_status=0 then Pass env
+  else Fail (Actions_helpers.mkreason
+    "expect" (String.concat " " commandline) exit_status)
+
+let run_expect_twice ocamlsrcdir input_file log env =
+  let corrected filename = Filename.make_filename filename "corrected" in
+  let first_run = run_expect_once ocamlsrcdir input_file false log env in
+  match first_run with
+    | Pass env1 ->
+      let intermediate_file = corrected input_file in
+      let second_run =
+        run_expect_once ocamlsrcdir intermediate_file true log env1 in
+      (match second_run with
+      | Pass env2 ->
+        let output_file = corrected intermediate_file in
+        let output_env = Environments.add_bindings
+        [
+          Builtin_variables.reference, input_file;
+          Builtin_variables.output, output_file
+        ] env2 in
+        Pass output_env
+      | Skip _ | Fail _ -> second_run
```

Comment:
This is minor, but I find the code more readable if the short cases are first (`Skip _ | Fail _` before `Pass env`). Otherwise when I read the short cases, I don't remember well what was being matched.